### DISCUSSION TITLE
How to edit my ZSH Git plugin alias that contains "$(current_branch)"?

### DISCUSSION THREAD

**Discussion by xuwhite** [ohmyzsh/ohmyzsh#12432] [Help]
I want to change "ggf" from `git push --force origin $(current_branch)` to `git push --force-with-lease origin $(current_branch)`, I know that "ggfl" exists, but still want to change it. 

Please help.

**Comment by ShreeChaturvedi** [comment]
`ggf` is a function in the git plugin, not a plain alias. Redefine it in `~/.zshrc` after Oh My Zsh loads (`source $ZSH/oh-my-zsh.sh`).

```zsh
function ggf() {
  local b
  [[ $# != 1 ]] && b="$(git_current_branch)"
  git push --force-with-lease origin "${b:-$1}"
}
compdef _git ggf=git-push
```

That keeps the same optional branch argument as the plugin. The plugin already ships `ggfl` as force-with-lease. Current plugin code uses `git_current_branch` (`current_branch` is only a compatibility name).

Do not edit `~/.oh-my-zsh/plugins/git/git.plugin.zsh`. Plugin updates will overwrite local edits.

Checked current `plugins/git/git.plugin.zsh` in `ohmyzsh/ohmyzsh`.
